### PR TITLE
Add rich session data

### PR DESCRIPTION
**Pull request by manuthecoder**
See #6712

### CODE REVIEW

**Review by sampoder — CHANGES_REQUESTED** [review]
Can we use `maps.hackclub.com`? Search the codebase for examples of where we've used it.

**Review by sampoder — CHANGES_REQUESTED** [review]
Thanks for that! A couple of things, geolocating for city is going to burn through all of our Geolocated API credits. Every time we create a user session, we save its longitude and latitude, so you can do `session.longitude` and `session.latitude`.

Secondly, for sessions that don't have an IP it makes the page look very imbalanced as there is a massive map on some sessions and nothing on others. Any thoughts on how we could fix that? Maybe there's a way we can design it in a way that the height doesn't change whether or not you have a map? Or we can render the entire world if we don't know? I'm not sure.

**Comment by manu-gurudath** [comment]
>...geolocating for city is going to burn through all of our Geolocated API credits...

What if HCB used [findip.net's](findip.net) API? It's free and unlimited. 

>  ... for sessions that don't have an IP it makes the page look very imbalanced ... Any thoughts on how we could fix that?

What if we showed an empty square with "No preview available," like how my app currently does:
image

I'm also formatting the map & placeholder with`aspect-ratio: 1 / 1` to ensure it remains the correct dimensions.

File: app/views/users/_user_session.html.erb [hunk 1]
```diff
@@ -22,6 +22,13 @@
       <% end %>
     <% end %>
   </span>
+  <% if session.latitude %>
+    <img class="w-full rounded mb-3 aspect-square" src="https://maps.hackclub.com/api/shipping?latitude=<%= session.latitude %>&longitude=<%= session.latitude %>" />
```
Comment (sampoder) [inline app/views/users/_user_session.html.erb]:
```suggestion
    <img class="w-full rounded mb-3 aspect-square" src="https://maps.hackclub.com/api/shipping?latitude=<%= session.latitude %>&longitude=<%= session.longitude %>" />
```

**Review by sampoder — APPROVED** [review]
LGTM!

**Pull request merged**